### ISSUE TITLE
Acts as tenant through a belongs to association

### ISSUE THREAD

**Issue by jimmypoulsen** [ErwinM/acts_as_tenant#293]
Hi.

We're currently implementing multi-tenancy in our single tenant application. Most of our application involves our `CaseFile` model. A case file has many offers, time entries and expenses. See the following model:

![account](https://user-images.githubusercontent.com/7967033/182136791-ae60214a-2be9-411e-b315-4824b2526be2.jpg)

Neither offers, time entries or expenses can exist without a case file so I want to avoid having the `account` association on each of these models. I only want to have it on `CaseFile`.

Is this possible?

**Comment by excid3** [comment]
I don't think we plan to support that. It's safer to have the `tenant_id` on each model because they can then be queried directly without the joins and just generally protects you since every record knows which tenant they belong to.

**Comment by excid3** [comment]
Actually, this may do what you wanted: https://github.com/ErwinM/acts_as_tenant/commit/a05d9a2e07dc396fd56c857e7934bef67bd9dbc9

**Comment by jimmypoulsen** [comment]
I came to the same conclusion because of the issue with using joins so I just implemented the `tenant_id` on all tenant records. I don't think [a05d9a2](https://github.com/ErwinM/acts_as_tenant/commit/a05d9a2e07dc396fd56c857e7934bef67bd9dbc9) would have solved my problem as I wasn't looking for a way to associate them using a `has_and_belongs_to_many` association. In my example a case file has many offers but an offer only belongs to one case file.
